libs/MobileSync/MobileSync/Classes/MobileSync.swift

```diff
@@ -0,0 +1,78 @@
+import Foundation
+
+/*
+ MobileSync.swift
+ MobileSync Swift Extensions
+ 
+ Copyright (c) 2019-present, salesforce.com, inc. All rights reserved.
+ 
+ Redistribution and use of this software in source and binary forms, with or without modification,
+ are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this list of conditions
+ and the following disclaimer.
+ * Redistributions in binary form must reproduce the above copyright notice, this list of
+ conditions and the following disclaimer in the documentation and/or other materials provided
+ with the distribution.
+ * Neither the name of salesforce.com, inc. nor the names of its contributors may be used to
+ endorse or promote products derived from this software without specific prior written
+ permission of salesforce.com, inc.
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+import Combine
+
+/// Errors that can be thrown using MobileSync
+public enum MobileSyncError: Error {
+    case error(_ error: Error?)
+    case unknown
+}
+
+extension SyncManager {
+    
+    /// Runs or reruns a sync.
+    /// - Parameter named: name of sync to run
+    /// - Parameter completionBlock: block invoked with Result<Bool, MobileSyncError>)
+    /// Note: boolean indicates completion/failure, error is only returned if the sync could not be started (e.g. invalid name)
+    public func runSync(named syncName: String, _ completionBlock: @escaping (Result<Bool, MobileSyncError>) -> Void) {
```

Comment:
Can't call it reSync even though signature is different.
I'm not sure about calling it runSync.

Reply:
What's wrong with runSync?

Reply:
Nothing. But we used "reSync" for years. But if I call it "reSync", swift finds it ambiguous because there is another reSync method that also take a name and completion block. The only difference is the completion block signature.

Reply:
@wmathurin We should used NS_REFINED_FOR_SWIFT for resync. Do we need resync to be available for swift consumption?

Reply:
Well reSync calls the update block several times as the sync runs. It can be useful for apps that want to show a progress bar etc.
This one calls the completion block only once when the sync completes or fails.
Maybe I should use a more descriptive name: reSyncNoUpdates?